dts/xtensa/intel/intel_adsp_ace20_lnl.dtsi

```diff
@@ -667,4 +687,52 @@
 			reg = <0x12>;
 		};
 	};
+
+	uaols {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		uaol13: uaol@d {
+			compatible = "intel,uaol-dai";
+			reg = <0x0d>;
+			link = <0>;
+			stream = <0>;
+			status = "okay";
+		};
+		uaol14: uaol@e {
+			compatible = "intel,uaol-dai";
+			reg = <0x0e>;
+			link = <0>;
+			stream = <1>;
+			status = "okay";
+		};
+		uaol15: uaol@f {
+			compatible = "intel,uaol-dai";
+			reg = <0x0f>;
+			link = <0>;
+			stream = <2>;
+			status = "okay";
+		};
+		uaol16: uaol@10 {
+			compatible = "intel,uaol-dai";
+			reg = <0x10>;
+			link = <0>;
+			stream = <3>;
+			status = "okay";
+		};
+		uaol29: uaol@1d {
+			compatible = "intel,uaol-dai";
+			reg = <0x1d>;
+			link = <0>;
+			stream = <4>;
+			status = "okay";
+		};
+		uaol30: uaol@1e {
+			compatible = "intel,uaol-dai";
+			reg = <0x1e>;
+			link = <0>;
+			stream = <5>;
+			status = "okay";
+		};
```

Comment:
These should be a child of `uaollink0` node.
From the DAI node the link device is the parent, so you don't need to fiddle with the link properties.

Reply:
Rearranged the code so that DAI nodes are children of UAOL link node(s). Thanks!